core/src/services/gcs/core.rs

```diff
@@ -318,16 +333,22 @@ impl GcsCore {
 
         let mut req = Request::put(&url);
 
+        if let Some(user_metadata) = args.user_metadata() {
```

Comment:
I'm not sure if this metadata should be set in the headers of presign requests. Any thoughts about this?

Reply:
Yes, I think it should. Presigned requests are just normal read/write requests, but signed in a different way.